### PR TITLE
Allow DataFrame.sort_values(by=) to select an index level

### LINKED ISSUE #14794: [BUG] sort_values fails on groupby aggregation
**Describe the bug**
This bug was found while playing around with the [One Billion Row Challenge](https://medium.com/coiled-hq/1brc-in-python-with-dask-3cdee6a56a2d).

**Steps/Code to reproduce bug**
```python
import cudf
import cupy as cp

# Generate some sample data similar to the data in 1BRC
stations = cudf.Series(['Ankara', 'Kampala', 'Tallinn', 'Gjoa Haven', 'Luanda', 'Cairo', 'Phnom Penh', 'Thessaloniki', 'Split', 'Palermo', 'Ouarzazate', 'Mandalay'])
df = cudf.DataFrame({        
        "station": cp.random.randint(0, len(stations)-1, 10_000_000),
        "measure": cp.random.normal(0, 10.0, 10_000_000)
    })
df.station = df.station.map(stations)

# Perform the grouby aggregation
df = df.groupby("station").agg(["min", "max", "mean"])
df.columns = df.columns.droplevel()

# Sort
df.sort_values("station") # <------------ Raises a KeyError

# It works in Pandas
df.to_pandas().sort_values("station")  # Works!
```

```pytb
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[10], line 1
----> 1 df.sort_values("station")

File /opt/conda/lib/python3.10/site-packages/cudf/core/indexed_frame.py:2459, in IndexedFrame.sort_values(self, by, axis, ascending, inplace, kind, na_position, ignore_index)
   2454     return self
   2456 # argsort the `by` column
   2457 out = self._gather(
   2458     GatherMap.from_column_unchecked(
-> 2459         self._get_columns_by_label(by)._get_sorted_inds(
   2460             ascending=ascending, na_position=na_position
   2461         ),
   2462         len(self),
   2463         nullify=False,
   2464     ),
   2465     keep_index=not ignore_index,
   2466 )
   2467 if (
   2468     isinstance(self, cudf.core.dataframe.DataFrame)
   2469     and self._data.multiindex
   2470 ):
   2471     out.columns = self._data.to_pandas_index()

File /opt/conda/lib/python3.10/site-packages/nvtx/nvtx.py:115, in annotate.__call__.<locals>.inner(*args, **kwargs)
    112 @wraps(func)
    113 def inner(*args, **kwargs):
    114     libnvtx_push_range(self.attributes, self.domain.handle)
--> 115     result = func(*args, **kwargs)
    116     libnvtx_pop_range(self.domain.handle)
    117     return result

File /opt/conda/lib/python3.10/site-packages/cudf/core/dataframe.py:1959, in DataFrame._get_columns_by_label(self, labels, downcast)
   1950 @_cudf_nvtx_annotate
   1951 def _get_columns_by_label(
   1952     self, labels, *, downcast=False
   1953 ) -> Self | Series:
   1954     """
   1955     Return columns of dataframe by `labels`
   1956 
   1957     If downcast is True, try and downcast from a DataFrame to a Series
   1958     """
-> 1959     ca = self._data.select_by_label(labels)
   1960     if downcast:
   1961         if is_scalar(labels):

File /opt/conda/lib/python3.10/site-packages/cudf/core/column_accessor.py:381, in ColumnAccessor.select_by_label(self, key)
    379     if any(isinstance(k, slice) for k in key):
    380         return self._select_by_label_with_wildcard(key)
--> 381 return self._select_by_label_grouped(key)

File /opt/conda/lib/python3.10/site-packages/cudf/core/column_accessor.py:536, in ColumnAccessor._select_by_label_grouped(self, key)
    535 def _select_by_label_grouped(self, key: Any) -> ColumnAccessor:
--> 536     result = self._grouped_data[key]
    537     if isinstance(result, cudf.core.column.ColumnBase):
    538         return self.__class__({key: result}, multiindex=self.multiindex)

KeyError: 'station'
```

**Expected behavior**
The same code works in Pandas/Numpy.

**Environment overview (please complete the following information)**

```bash
docker run --gpus all --pull always --rm -it \
    --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 \
    -p 8888:8888 -p 8787:8787 -p 8786:8786 \
    rapidsai/notebooks:23.12-cuda12.0-py3.10
```

**Additional context**

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
closes #14794

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -450,6 +451,15 @@ def __getitem__(self, index):
             return self.start + index * self.step
         return self._as_int_index()[index]
 
+    def _get_columns_by_label(self, labels) -> Index:
+        # used in .sort_values
+        if isinstance(labels, Hashable) and labels == self.name:
+            return self._as_int_index()
+        elif list(self.names) == list(labels):
```
Comment (Matt711) [inline python/cudf/cudf/core/index.py:458]: I think CI is failing on this example `df = cudf.DataFrame([1], columns=cudf.Index(np.array([1], dtype=np.int8))).sort_values(0)`. You get a TypeError from `list(labels)`.

Should this check if `labels` are in `self.names`?

Comment (mroeschke) [reply python/cudf/cudf/core/index.py]: Thanks I was able to fix this.

The issue here is that `labels` can be scalar-like or list-like so the equality matching needs to fit both cases

**Review by Matt711 — APPROVED** [review]
LGTM

**Pull request merged**